altimeter_ms5607_spi/altimeter_ms5607_spi.cc

```diff
@@ -0,0 +1,108 @@
+#include "altimeter_ms5607_spi.h"
+
+AltimeterMs5607Spi::AltimeterMs5607Spi(SPI_HandleTypeDef *hspi, GPIO_TypeDef *csPort, uint16_t csPin, GPIO_TypeDef *misoPort, uint16_t misoPin, double seaLevelPressure)
+    : _hspi(hspi), _csPort(csPort), _csPin(csPin), _misoPort(misoPort), _misoPin(misoPin), _seaLevelPressure(seaLevelPressure) {}
+
+int AltimeterMs5607Spi::Reset() {
+    HAL_GPIO_WritePin(_csPort, _csPin, GPIO_PIN_RESET);
+    // reset, opcode 0x1E
+    if (HAL_SPI_Transmit(_hspi, (uint8_t *)(const uint8_t[]){0x1E}, 1, SERIAL_TIMEOUT) != HAL_OK) return -1;
+    HAL_GPIO_WritePin(_csPort, _csPin, GPIO_PIN_SET);
+    return 0;
+}
+
+int AltimeterMs5607Spi::Init() {
+    for (int i = 0; i < 6; i++) {
+        uint8_t buffer[2] = {0};
+        HAL_GPIO_WritePin(_csPort, _csPin, GPIO_PIN_RESET);
+        // PROM read, opcode 0b1010xxx0
+        uint8_t command[1] = {(uint8_t)(0b10100000 | ((i + 1) << 1))};
+        if (HAL_SPI_Transmit(_hspi, command, 1, SERIAL_TIMEOUT) != HAL_OK) return -1;
+        if (HAL_SPI_Receive(_hspi, buffer, 2, SERIAL_TIMEOUT) != HAL_OK) return -1;
+        _coefficients[i] = ((uint16_t)buffer[0] << 8) | (uint16_t)buffer[1];
+        HAL_GPIO_WritePin(_csPort, _csPin, GPIO_PIN_SET);
+    }
+    return 0;
+
+    // TODO: check CRC
+}
+
+AltimeterMs5607Spi::Data AltimeterMs5607Spi::Read() {
+    AltimeterMs5607Spi::Data data;
+
+    uint32_t d1 = 0;
+    uint32_t d2 = 0;
+
+    {
```

Comment:
What's with these curly-brace blocks? This is pretty unusual. I would just separate them by spaces or comments from eachother but not have curly braces like this.

Reply:
Added so that some of the temp variables only exist within the scope, but doesn't help with readability. Will remove.